spyder/widgets/sourcecode/tests/test_folding.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Tests for the autoindent features
+"""
+
+# Third party imports
+from qtpy.QtGui import QTextCursor
+import pytest
+
+# Local imports
+from spyder.utils.qthelpers import qapplication
+from spyder.widgets.sourcecode.codeeditor import CodeEditor
+from spyder.widgets.sourcecode.api.folding import print_tree
+
+
+# --- Fixtures
+# -----------------------------------------------------------------------------
+@pytest.fixture()
+def get_fold_levels():
+    """setup editor and return fold levels."""
+
+    def get_fold_levels():
```

Comment:
Why this nested function with the same name as the outer? Can't we get rid of it?